### PR TITLE
Stop throwing an exception when ScopeManagerShim activate a null Span

### PR DESCRIPTION
**Pull request by shawnsong**
Details are described in: https://github.com/open-telemetry/opentelemetry-java/issues/3952

A scope is returned to make sure the context is restored.

### CODE REVIEW

**Comment by shawnsong** [comment]
Hi @carlosalberto, do you mind taking a quick look at this PR? This is my first attempt :) Thanks.

File: opentracing-shim/src/test/java/io/opentelemetry/opentracingshim/TracerShimTest.java [hunk 1]
```diff
@@ -80,6 +80,25 @@ void activateSpan() {
     assertThat(io.opentelemetry.api.baggage.Baggage.current()).isSameAs(EMPTY_BAGGAGE);
   }
 
+  @Test
+  void activateNullSpan() {
```
Comment (carlosalberto) [inline opentracing-shim/src/test/java/io/opentelemetry/opentracingshim/TracerShimTest.java:84]: What about starting the test with an actual active `Span`, _and then_ you activate/test the null value?

For testing purposes it's the same, but I get the feeling that changing it that way could make the test more readable.

File: opentracing-shim/src/main/java/io/opentelemetry/opentracingshim/ScopeManagerShim.java [hunk 2]
```diff
@@ -44,7 +44,12 @@ public Span activeSpan() {
 
   @Override
   @SuppressWarnings("MustBeClosedChecker")
-  public Scope activate(Span span) {
+  public Scope activate(@Nullable Span span) {
+    if (span == null) {
+      SpanShim spanShim =
```
Comment (carlosalberto) [inline opentracing-shim/src/main/java/io/opentelemetry/opentracingshim/ScopeManagerShim.java:49]: I think we could cache a final private `SpanShim` instance that simply wraps OTel `Span.getInvalid()`, and we can re-use that for all invocations:

```java
if (span == null) {
  return new ScopeShim(Context.current().with(NOP_SPANSHIM).makeCurrent());
}
```

**Review by jkwatson — APPROVED** [review]

**Review by anuraaga — APPROVED** [review]

**Comment by anuraaga** [comment]
Thanks!

**Pull request merged**